toc/rfc/rfc-draft-efficient-modern-vm-sizes.md

```diff
@@ -0,0 +1,118 @@
+the default VM sizes for these workloads. We will choose VM sizes that are
+either cheaper because they are newer generation, they provide a smaller/finer
+tuned amount of hardware, or they have burstable cores for appropriate
+workloads.
+
+**Note: This is a breaking change, especially for [Reserved
+Instance](https://aws.amazon.com/ec2/pricing/reserved-instances/) holders.
+Anyone using the default VM sizes for a high-performance or production CF should
+consider customizing their VM sizes. In addition, the Diego cells have been
+reduced from 26-64GB of memory to a consistent 16GB by default, so please
+consider how many you need or resize them up.**.
+
+We believe it is acceptable to make this type of change because we expect
+production deployments of CF to customize their VM types anyway, and we cannot
+prevent progress just to help RI buyers.
+
+## Problem
+
+We'd like to save the CFF, and anyone who deploys default CF configuration,
+significant money in operating BOSH and CF deployment. The current deployment
+costs for CF-d are so expensive (for example on AWS) that our limited foundation
+CI budget will be consumed too quickly. In addition, our VM type usages are in
+need of a refresh to match the newest IaaS offerings for both performance and
+cost.
+
+## Proposal
+
+### Important decisions:
+* VMs are allowed to be burstable, in minimal use cases.
+* Diego cells should be smaller and consistent at 16GB of RAM, rather then the
+  old 30-50GB ranges.
+* Modern (as of 2021) instance types should be used.
+
+### VM Size definitions
+We'd like to further clarify the meaning of the default VM sizes for CF-D,
+BOSH-D, and Jumpbox-D as follows:
+
+| Deployment Used | VM Size Name | Definition |
+| --- | --- | --- |
+| bosh | bosh | A VM that MUST have at least 4 GB of memory and 1 CPU. The CPU MAY be burstable, since the director is often idle. Since UAA and/or Credhub are common add-ons that add significant memory, an ops-file with 8GB RAM and more CPU MAY be provided. |
+| jumpbox | jumpbox | A minimal VM with at least 1GB of memory and 1 CPU; It SHOULD be burstable. |
+| cf | minimal | A minimal VM that SHOULD have 2GB+ of memory and 1+ CPU; It SHOULD be burstable. |
+| cf | small | A default VM size for larger processes that SHOULD have 4GB of memory with 1 or 2 CPUs. It MAY be burstable if the VM can sustain baseline usage of at least 75% usage on a vCPU. |
+| cf | small-highmem | A default app running VM size that SHOULD have 16GB+ of memory and 2 CPUs. It MAY be burstable since app workloads can be burstable. These SHOULD have a higher memory-to-CPU ratio to optimize cost and fit more applications per Diego Cell. |
+| bosh | compilation | A high-compute, short-running VM pool that SHOULD have 2+ CPUs with at least 4GB of RAM and SHOULD NOT be burstable. This pool runs typically for 5-20 minutes, so it SHOULD be 5 nodes for per-minute or per-second billing or 3 nodes for hourly billing. |
```

Comment:
You mention 3 nodes for hourly billing, but that doesn't apply to any of the IaaSs listed below. What is this relevant for?

Reply:
I mean... I _thought_ AWS and Azure were hourly billing so we'd want to run fewer VMs for longer, but apparently that's all been fixed. Will drop it!